gravitee-am-authenticator/gravitee-am-authenticator-api/src/main/java/io/gravitee/am/authenticator/api/AuthenticatorHandler.java

```diff
@@ -34,7 +36,11 @@ public class AuthenticatorHandler implements Handler<RoutingContext> {
     public void handle(RoutingContext routingContext) {
         authenticator.authenticate(routingContext)
                 .doOnSuccess(user -> auditService.report(authenticator.successAuditLog(routingContext, user)))
-                .doOnError(err -> auditService.report(authenticator.failedAuditLog(routingContext, err)))
+                .doOnError(err -> {
+                    var auditEvent = authenticator.failedAuditLog(routingContext, err);
+                    auditEvent.principal(setAuditUser(err));
```

Comment:
suggestion: why not doing this setAuditUser into the failedAuditLog method so the Authenticator keep the full logic of the Audit creation?